libvast/src/system/node.cpp

```diff
@@ -268,13 +268,7 @@ caf::message dump_command(const invocation& inv, caf::actor_system&) {
                                              "type-registry"));
   caf::error request_error = caf::none;
   auto rp = self->make_response_promise();
-  // The overload for 'request(...)' taking a 'std::chrono::duration' does not
-  // respect the specified message priority, so we convert to 'caf::duration'
-  // by hand.
-  const auto timeout = caf::duration{defaults::system::initial_request_timeout};
-  self
-    ->request<caf::message_priority::high>(type_registry, timeout, atom::get_v,
```

Comment:
Why are we not using a timeout anymore here? (same question applies to the other places below where it was removed)

Reply:
Because this is always running with the same process, so using a timeout just doesn't make sense here. I did not understand why we were using one in the first place.

Reply:
I strongly disagree. Timeouts are not just for IPC. Every request should have a timeout, ideally as low as possible. If a request times out, then ideally we try again, but keeping requests timeouts low dramatically increases the responsiveness of a system.

There's a great talk from the WhatsApp team that doubled down on this point in their Erlang stack. I can't find it right now though.

Reply:
I understand this generally and I've actually seen that talk—it's a great one—, but CAF is making this really hard in practice. Specifically, I am thinking of two issues:

1. Streaming and regular messages have different queues, with streaming always being preferred. This can lead to requests not going through when using timeouts when an actor is busy while streaming. 

3. CAF's response promises cannot by design know when they're cancelled, so the work of the request will continue even throughout multiple stages without us ever being able to know whether the source of the request is still alive. E.g., if you have a request from actor A going to actor B that delegates to actor C, then B will always delegate to C regardless of whether the request from A to B timed out. This is a design flaw in CAF that would cause repeated requests to do repeated work regardless of whether or not they were cancelled.

Reply:
Good points.

In my view, these issues make it harder to find the right *tight* lower bound on the timeout, but it doesn't invalidate the low-timeout approach in principle.

I can see the CAF Streaming throws a wrench in the timeout computation, as high load can cause infinite delays. Is there a fix in CAF to use the same queue? In theory, that shouldn't be hard.

The response cancellation seems less of an issue in this case. It does the right thing by entering the error case when no *response* has been received in the provided time frame. You're asking for "exactly-once" semantics of the work elicited by the request. I think that requirement is not needed for the common case, and needs a different primitive when it's required.

Reply:
I would like to decouple the timeout discussion from this PR. If we're doing this then we need to discuss it more generally, but having a few timeouts of seemingly random durations throughout VAST hurts developers in debugging because they lead to very unexpected failures if VAST is really busy, especially with streaming and cascading requests.

This will hopefully get better with CAF 0.19, but we have yet to see the new streaming implementation on top of flows, so who really knows. For now, this is not something that we can or should do, and this also makes it so we should not re-open this discussion before we upgraded to CAF 0.19.

Reply:
Agreeing here, half-heartedly. I'll leave it to @lava to resolve this conversation, assuming you are okay with the approach taken.